File: mobile/android/app/src/main/java/io/meshtech/koasta/net/model/UserProfile.kt
package io.meshtech.koasta.net.model

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class UserProfile(
  val registrationId: String,
  val wantAdvertising: Boolean,
  var votedVenueIds: List<Int>,
  val firstName: String?,
  val lastName: String?
)

File: mobile/ios/src/data/models/user-profile.swift
import Foundation

struct UserProfile: Codable {
  let registrationId: String
  let wantAdvertising: Bool
  let votedVenueIds: [Int]
  let firstName: String?
  let lastName: String?

  func withVotedVenueId(_ venueId: Int) -> UserProfile {
    var ids = votedVenueIds
    ids.append(venueId)
    return UserProfile(registrationId: registrationId, wantAdvertising: wantAdvertising, votedVenueIds: ids, firstName: firstName, lastName: lastName)
  }
}
